decimal64.go

```diff
@@ -305,3 +447,28 @@ func (d Decimal64) Sign() int {
 func (d Decimal64) Signbit() bool {
 	return d.bits>>63 == 1
 }
+
+//size returns the magnitude (number of digits) of a uint64.
+func size(n uint64) int {
+	numBits := 64 - bits.LeadingZeros64(n)
+	numDigits := numBits * 3 / 10
+	if n < powersOf10[numDigits] {
+		return numDigits
+	}
+	return numDigits + 1
+}
+
+// propagateNan returns true if flavor1 should be propagated else false if flavor2 should be propagated
+func propagateNan(dp, ep *decParts) *Decimal64 {
+	if dp.fl == flSNaN {
```

Comment:
```go
switch dp.fl {
```

Reply:
behaviour between sNans and qNans are the same now